src/Microsoft.ML.Data/Utilities/StreamUtils.cs

```diff
@@ -174,5 +174,40 @@ private static string[] Expand(string pattern)
             return matchList.ToArray();
         }
 #endif
+
+        public static string ReadEntry(this TextReader sr)
+        {
+            string strReturn = string.Empty;
+
+            // get first bit
+            strReturn += sr.ReadLine();
+
+            // And get more lines until the number of quotes is even
+            while (strReturn.GetNumberOf("\"").IsOdd())
+            {
+                string strNow = sr.ReadLine();
+                strReturn += strNow;
+            }
+
+            // Then return what we've gotten
+            if (strReturn == string.Empty)
+            {
+                return null;
+            }
+            else
+            {
+                return strReturn;
+            }
+        }
+
+        public static int GetNumberOf(this string s, string strSearchString)
```

Comment:
I would just leave this as a static method instead of an extension method. It's easier to read that way.

Reply:
I would agree, especially with completion for unimported extension methods coming from Roslyn. Specialized extension methods for common types will be uncomfortably noisy during development.